pandas/_libs/src/inference.pyx

```diff
@@ -222,14 +223,16 @@ cdef _try_infer_map(v):
     return None
 
 
-def infer_dtype(object value):
+def infer_dtype(object value, bint skipna=False):
     """
     Effeciently infer the type of a passed val, or list-like
     array of values. Return a string describing the type.
 
     Parameters
     ----------
     value : scalar, list, ndarray, or pandas type
+    skipna : bool, default False
+        Ignore NaN values when inferring the type.
```

Comment:
Minor thing: I would prefer that you still mention that we will change the default to `True` in the future, even if we don't deprecate now.  What do you think?